feat: add skip blur option

mrpmohiburrahman opened the pull request:
<!-- Please provide enough information so that others can review your pull request. -->
<!-- Keep pull requests small and focused on a single change. -->

### Summary

Resolves: https://github.com/callstack/react-native-testing-library/issues/1677

<!-- What existing problem does the pull request solve? Can you solve the issue with a different approach? -->

### Test plan

<!-- List the steps with which we can test this change. Provide screenshots if this changes anything visual. -->
All tests pass

mdjastrzebski commented on src/user-event/type/type.ts:
```diff
@@ -69,7 +70,9 @@ export async function type(
   dispatchEvent(element, 'endEditing', EventBuilder.TextInput.endEditing(finalText));
 
-  dispatchEvent(element, 'blur', EventBuilder.Common.blur());
+  if (!options?.skipBlur) {
```
skip blur option should disable both endEditing and blur together, as in in native platforms (iOS, android) they always happen together.

mrpmohiburrahman replied:
Made the changes on the latest commit.

mdjastrzebski reviewed (changes requested):
I've left comment about events affected by this flag. Please add also documentation entry for the option.

mdjastrzebski commented:
Document to update: https://github.com/callstack/react-native-testing-library/blob/main/website/docs/12.x/docs/api/events/user-event.mdx.

Please also add units test(s) for behavior with `skipBlur` option set.

mrpmohiburrahman commented:
> Document to update: https://github.com/callstack/react-native-testing-library/blob/main/website/docs/12.x/docs/api/events/user-event.mdx.
> 
> Please also add units test(s) for behavior with `skipBlur` option set.

Hi, @mdjastrzebski , 

I have updated doc and added unit test.
please, check.
Also note that, I am very new to this.

mdjastrzebski reviewed (approved):
Ready to merge! @mrpmohiburrahman thank you for this contribution.
